ledger/apply/keyreg.go

```diff
@@ -62,9 +65,11 @@ func Keyreg(keyreg transactions.KeyregTxnFields, header transactions.Header, bal
 		record.VoteFirstValid = 0
 		record.VoteLastValid = 0
 		record.VoteKeyDilution = 0
+		if params.EnableStateProofKeyregCheck {
```

Comment:
I think we want to move this up to where VoteID and SelectionID are set?

Reply:
well, the other public keys are updated there with the comment
```
    // Update the registered keys and mark account as online
    // (or, if the voting or selection keys are zero, offline/not-participating)
    record.VoteID = keyreg.VotePK
    record.SelectionID = keyreg.SelectionPK
```
so arguably the reason we have this bug is because the new voting public key (StateProofID) was not added here section originally, and was only added to one of the if/else sections.

Plus you get to deduplicate code — in both cases you are setting `record.StateProofID = keyreg.StateProofPK`

Reply:
I'm confused now because I'm only setting `record.StateProofID = keyreg.StateProofPK` in one case

Reply:
Also what if someone submits a txn with offline accounts but not empty stateproofpk?

Reply:
setting `record.StateProofID` when `keyreg.StateProofPK` is empty is the same thing as setting it to `merklesignature.Verifier{}`. That's what it means in the comment "Update the registered keys ... or, if the voting or selection keys are zero, offline/not-participating"

Reply:
fixed in e43e25156b233c9c8c5c764282f3e8e42022d780